### PR TITLE
[Trivial] Format "and" as instruction op code

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: docs/assembly.rst [hunk 1]
```diff
@@ -346,7 +346,7 @@ Literals
 
 You can use integer constants by typing them in decimal or hexadecimal notation and an
 appropriate ``PUSHi`` instruction will automatically be generated. The following creates code
-to add 2 and 3 resulting in 5 and then computes the bitwise and with the string "abc".
+to add 2 and 3 resulting in 5 and then computes the bitwise ``and`` with the string "abc".
```
Comment (axic) [inline docs/assembly.rst]: I think uppercase AND would be more fitting (the push is uppercase too), or italic `bitwise and`.

**Comment by Marenz** [comment]
* Made `and` UPPERCASE
* Added commit to rephrase opcode return behavior description

File: docs/assembly.rst [hunk 2]
```diff
@@ -159,8 +159,9 @@ following list can be used as a reference of its opcodes.
 
 If an opcode takes arguments (always from the top of the stack), they are given in parentheses.
 Note that the order of arguments can be seen to be reversed in non-functional style (explained below).
-Opcodes marked with ``-`` do not push an item onto the stack, those marked with ``*`` are
-special and all others push exactly one item onto the stack.
+If an opcode returns a result, it is done by pushing it on the stack.
+Opcodes marked with ``-`` don't return a result (don't push an item onto the stack),
```
Comment (chriseth) [inline docs/assembly.rst]:
```suggestion
Opcodes marked with ``-`` do not return a result (do not push an item onto the stack),
```

**Comment by Marenz** [comment]
* Changed `don't` -> `do not`
* Added in-depth explanation of opcode sequence

File: docs/assembly.rst [hunk 3]
```diff
@@ -159,8 +159,9 @@ following list can be used as a reference of its opcodes.
 
 If an opcode takes arguments (always from the top of the stack), they are given in parentheses.
 Note that the order of arguments can be seen to be reversed in non-functional style (explained below).
-Opcodes marked with ``-`` do not push an item onto the stack, those marked with ``*`` are
-special and all others push exactly one item onto the stack.
+If an opcode returns a result, it is done by pushing it on the stack.
```
Comment (chriseth) [inline docs/assembly.rst]: This sounds like I have to know whether an opcode returns a result or not in order to read the table below. What about removing this sentence and instead changing the next to end in `and all others push exactly one item onto the stack (their "return value")` or something like that.

File: docs/assembly.rst [hunk 4]
```diff
@@ -366,6 +367,21 @@ arguments for certain opcodes are. In the following example,
 
     3 0x80 mload add 0x80 mstore
 
+Going through this step by step, from left to right:
+
+* ``3`` is put on the top of the stack.
+* ``0x80`` is put on the top of the stack.
+* ``mload`` is called, which takes the top most element as argument,
```
Comment (chriseth) [inline docs/assembly.rst]:
```suggestion
* ``mload`` is called, which takes the topmost element as argument,
```

File: docs/assembly.rst [hunk 5]
```diff
@@ -366,6 +367,21 @@ arguments for certain opcodes are. In the following example,
 
     3 0x80 mload add 0x80 mstore
 
+Going through this step by step, from left to right:
```
Comment (chriseth) [inline docs/assembly.rst]: I think this goes a little too much in detail for this section (which is just there to explain the benefits of functional style).

Perhaps we could add it to the section that introduces the EVM?

**Comment by chriseth** [comment]
Sorry, but documentation changes are never trivial ;)

Perhaps you could single out some of the easier changes into their own PR?

**Comment by Marenz** [comment]
Alright, removed the controversial commit for now and adjusted the rephrasing one

**Review by chriseth — APPROVED** [review]

**Pull request merged**